src/validations/passport-gated/index.ts

```diff
@@ -27,46 +49,145 @@ function hasValidIssuanceAndExpiration(credential, proposalTs) {
   return false;
 }
 
+function hasStampCredential(stampId: string, credentials: Array<string>) {
+  const stamp = stampCredentials.find((stamp) => stamp.id === stampId);
+  if (!stamp) {
+    console.log('[passport] Stamp not supported', stampId);
+    throw new Error('Stamp not supported');
+  }
+  return credentials.some((credential) =>
+    stamp.credentials.includes(credential)
+  );
+}
+
+async function validateStamps(
+  currentAddress: string,
+  operator: string,
+  proposalTs: string,
+  requiredStamps: Array<string> = []
+): Promise<boolean> {
+  if (requiredStamps.length === 0) return true;
+
+  const stampsResponse = await fetch(GET_PASSPORT_STAMPS_URI + currentAddress, {
+    headers
+  });
+  const stampsData = await stampsResponse.json();
+
+  if (!stampsData?.items) {
+    console.log('[passport] Stamps Unknown error', stampsData);
+    throw new Error('Unknown error');
+  }
+  if (stampsData.items.length === 0) return false;
+
+  // check expiration for all stamps
+  const validStamps = stampsData.items
+    .filter((stamp: any) =>
+      hasValidIssuanceAndExpiration(stamp.credential, proposalTs)
+    )
+    .map((stamp: any) => stamp.credential.credentialSubject.provider);
+
+  if (operator === 'AND') {
+    return requiredStamps.every((stampId) =>
+      hasStampCredential(stampId, validStamps)
+    );
+  } else if (operator === 'OR') {
+    return requiredStamps.some((stampId) =>
+      hasStampCredential(stampId, validStamps)
+    );
+  }
+  return false;
+}
+
+function evalPassportScore(scoreData: any, minimumThreshold = 0): boolean {
+  // scoreData.evidence?.type === 'ThresholdScoreCheck' -> Returned if using Boolean Unique Humanity Scorer (should not be used)
+  if (scoreData.evidence?.type === 'ThresholdScoreCheck') {
+    return (
+      Number(scoreData.evidence.rawScore) > Number(scoreData.evidence.threshold)
+    );
+  }
+  // scoreData.score -> Returned if using Unique Humanity Score
+  return Number(scoreData.score) > minimumThreshold;
```

Comment:
Threshold should include minimum value also right?? like if `scoreThreshold` is `17.62`, any validation with a score of `17.61` or lower would be considered invalid, while any validation with a score of `17.62` or higher would be considered valid. no?

here it is checking from `17.63` and more to be valid

```suggestion
  return Number(scoreData.score) >= minimumThreshold;
```